### PR TITLE
[AAE-10904] Avoid technical info leaks in HTTP error responses

### PR DESCRIPTION
**Pull request by matteo4diani**

### CODE REVIEW

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/main/java/org/activiti/cloud/services/modeling/rest/controller/error/attributes/ErrorAttributesModelValidationErrorsBuilder.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        if (!collectedErrors.isEmpty()) {
+            errorAttributes.put(ERRORS,
+                collectedErrors);
+        }
+
+        return errorAttributes;
+    }
+
+    private Stream<ModelValidationError> resolveSemanticErrors(Throwable error,
+                                                               Map<String, Object> errorAttributes) {
```
Comment (erdemedeiros) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/main/java/org/activiti/cloud/services/modeling/rest/controller/error/attributes/ErrorAttributesModelValidationErrorsBuilder.java]: The parameter `errorAttributes` is never used in this method, so you can remove it.

File: activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    @ValueSource(strings = {
+        "An exception occurred @[java.lang.utils]",
+        "Could not serialize @[javax.validation]",
+        "Missing bean @[org.springframework]",
+        "Error at com.google.guava",
+        "Error at io.zipkin.brave",
+        "Error at org._int.obscurepackage"
+    })
+    void should_replaceMessageWithErrorNotDisclosed_when_matchesBlacklistedItems(String message) {
+        ErrorAttributesCustomizer sanitizer = getNewSanitizer();
```
Comment (erdemedeiros) [inline activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java]: You can move `sanitizer` at the field level and initialize it there. JUnit will create a new instance of it for each test in the suite.

File: activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        assertThat(errorAttributes).containsEntry(ErrorAttributesMessageSanitizer.MESSAGE, message);
+    }
+
+    private static ErrorAttributesCustomizer getNewSanitizer() {
+        return new ErrorAttributesMessageSanitizer();
+    }
+
+    private static Map<String, Object> getNewErrorAttributes() {
+        return new HashMap<>();
+    }
```
Comment (erdemedeiros) [inline activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java]: I don't think this methods are needed, you can use directly `new`.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Good job @matteo4diani ! There are only some small details to check and I think it would be nice to have an integration test for it as well. I think you can add one in [QueryProcessInstancesEntityIT](https://github.com/Activiti/activiti-cloud/blob/develop/activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java).

**Review by erdemedeiros — APPROVED** [review]
LGTM!

File: activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java (hunk 3 above)
Comment (matteo4diani) [reply activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java]: Done in [56a57f4](https://github.com/Activiti/activiti-cloud/pull/905/commits/56a57f4c8040e95f9c1eb6a2a49979d0cdde2d16)

File: activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java (hunk 2 above)
Comment (matteo4diani) [reply activiti-cloud-service-common/activiti-cloud-service-common-config/src/test/java/org/activiti/cloud/common/error/attributes/ErrorAttributesMessageSanitizerTest.java]: Done in [56a57f4](https://github.com/Activiti/activiti-cloud/pull/905/commits/56a57f4c8040e95f9c1eb6a2a49979d0cdde2d16)

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/main/java/org/activiti/cloud/services/modeling/rest/controller/error/attributes/ErrorAttributesModelValidationErrorsBuilder.java (hunk 1 above)
Comment (matteo4diani) [reply activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/main/java/org/activiti/cloud/services/modeling/rest/controller/error/attributes/ErrorAttributesModelValidationErrorsBuilder.java]: Done in [56a57f4](https://github.com/Activiti/activiti-cloud/pull/905/commits/56a57f4c8040e95f9c1eb6a2a49979d0cdde2d16)

**Pull request merged**